workflow/controller/controller_test.go

```diff
@@ -93,3 +119,23 @@ func makePodsRunning(t *testing.T, kubeclientset kubernetes.Interface, namespace
 		_, _ = podcs.Update(&pod)
 	}
 }
+
+func TestAddingWorkflowDefaultValueIfValueNotExist(t *testing.T) {
```

Comment:
I think we should probably have a few different tests here to make sure we understand how it'll work in different ways

Reply:
Sure I will add more tests :)

Reply:
looks good